pom.xml

```diff
@@ -1,70 +1,229 @@
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <version>2.12.4</version>
+                        <configuration>
+                            <skipTests>true</skipTests>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+
+        <profile>
+            <id>mpa</id>
+            <dependencies>
+                <dependency>
+                    <groupId>ch.ethz.rtc.kernel</groupId>
+                    <artifactId>rtc</artifactId>
+                    <scope>system</scope>
+                    <systemPath>${project.basedir}/lib/rtc.jar</systemPath>
+                    <version>1.2.beta.100</version>
+                </dependency>
+            </dependencies>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>build-helper-maven-plugin</artifactId>
+                        <version>1.7</version>
+                        <executions>
+                            <execution>
+                                <id>add-source</id>
+                                <phase>generate-sources</phase>
+                                <goals>
+                                    <goal>add-source</goal>
+                                </goals>
+                                <configuration>
+                                    <sources>
+                                        <source>src/mpa_ext</source>
+                                    </sources>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+
+        <profile>
+            <id>exp</id>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.codehaus.mojo</groupId>
+                        <artifactId>build-helper-maven-plugin</artifactId>
+                        <version>1.7</version>
+                        <executions>
+                            <execution>
+                                <id>add-source</id>
+                                <phase>generate-sources</phase>
+                                <goals>
+                                    <goal>add-source</goal>
+                                </goals>
+                                <configuration>
+                                    <sources>
+                                        <source>src/experiments</source>
+                                    </sources>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
```

Comment:
Will we have the possibility to have separated junit tests for each project or just the common tests containing functional tests can be used for this reason?

Reply:
Regarding the tests: At the moment we only have the possibility to run the functional tests for the whole project. Or am I mistaken? For future purposes it should be easy to add tests for the respective repositories and adding a e.g. `mvn package -P mpa_tests` profile for them.

Reply:
And how will we add them as submodule? src/test/java? Should not we add functional tests to src/functional/java?

Reply:
e.g. `src/mpa_tests/java`. But it's only a suggestion. Not sure how the best way look like.

Reply:
Yes, unit tests could have the test_[module name]/java to be consistent with Maven. Actually the junit tests and functional tests are already separated (unit in the corresponding project, functional in separate project). Perhaps we could prepare this in the maven as it is just a maven config issue. We could also leave it for the future as there are no unit tests at all at the moment.

Reply:
Where is the second repo? I only have access to `NetCal/DiscoDNC_tests` containing the tests I already included.

Reply:
Can we create functional test extensions?
Currently, we cannot run the functional tests for the DiscoDNC core only. The reason can be found in DncTestMethodSources where the use of the MPA RTC curve backend is hardcoded into the test parameter generation (createParameters method).

Reply:
However in the v2.5 branch, there is no `src/test` anymore. @sbondorf that's what I meant with adding a `src/tests_mpa` directory and moving the mpa parts of the tests there. Those tests can then be added via a profile.

Reply:
These are the same, just with a different configuration option for the existing tests. I would not know what to move. I rather feel the method source requires adaptation, e.g., taking a list of parameters that tell it the additional configuration options like MPA RTC curve backend. But I would not know how to pass that from the Maven profile.

Reply:
Why tests_mpa contain functional tests? The tests project at the moment contains only functional tests. I would add them to src/functional_test/java. src/tests_mpa should reference DiscoDNCext_MPARTC src/test/java. This way you separate unit and functional tests.

Reply:
Separation of different kinds of tests sounds reasonable to me. @phschon ?

Reply:
I would propose to solve the test discussion in a seperate issue, since it is an extension of this PR.  Thus in a first step I would merge the PR and include all tests in a single submodule and afterwards split the test repo and add maven profiles.

Reply:
But yes, in general the suggestions sounds promising.